device/mellanox/x86_64-mlnx_msn4600c-r0/Mellanox-SN4600C-C64/qos.json.j2

```diff
@@ -164,8 +210,67 @@
             "6": "6",
             "7": "0"
         }
+{% endif %}
+    },
+{%- endmacro %}
+{%- macro generate_tc_to_queue_map() %}
+    "TC_TO_QUEUE_MAP": {
+{% if (('type' in DEVICE_METADATA['localhost'] and DEVICE_METADATA['localhost']['type'] == 'ToRRouter') and
+       ('subtype' in DEVICE_METADATA['localhost'] and DEVICE_METADATA['localhost']['subtype'] == 'DualToR')) %}
+        "AZURE_TUNNEL": {
+            "0": "0",
+            "1": "1",
+            "2": "2",
+            "3": "2",
+            "4": "6",
+            "5": "5",
+            "6": "6",
+            "7": "7",
+            "8": "1"
+        },
+        "AZURE": {
+            "0": "0",
+            "1": "1",
+            "2": "2",
+            "3": "3",
+            "4": "4",
+            "5": "5",
+            "6": "6",
+            "7": "7",
+            "8": "1"
+        }
+{% else %}
+        "AZURE": {
+            "0": "0",
+            "1": "1",
+            "2": "2",
+            "3": "3",
+            "4": "4",
+            "5": "5",
+            "6": "6",
+            "7": "7"
+        }
+{% endif %}
+    },
+{%- endmacro %}
+{% if (('type' in DEVICE_METADATA['localhost'] and DEVICE_METADATA['localhost']['type'] == 'ToRRouter') and
+       ('subtype' in DEVICE_METADATA['localhost'] and DEVICE_METADATA['localhost']['subtype'] == 'DualToR')) %}
+{%- macro generate_tc_to_dscp_map() %}
+    "TC_TO_DSCP_MAP": {
+        "AZURE_TUNNEL": {
+            "0": "8",
+            "1": "0",
+            "2": "2",
+            "3": "2",
+            "4": "6",
+            "5": "46",
+            "6": "6",
+            "7": "48",
+            "8": "33"
```

Comment:
Since the `tc_mapped_from_dscp_33` is a variable, is it better to make `8` also a variable here?

Reply:
And should we map TC `2` and TC `6` to `0`? Even if we don't have DSCP values being mapped to TC `2` and `6`, mapping 2&6 to DSCP `0` should me more reasonable.

Reply:
> And should we map TC `2` and TC `6` to `0`? Even if we don't have DSCP values being mapped to TC `2` and `6`, mapping 2&6 to DSCP `0` should me more reasonable.

As discussed offline, we have to map TC `2` and `6` to DSCP `2` and `6` so that SAI will generate the correct settings to SDK/ASIC